### PR TITLE
Add a way to insert new items at the bottom of the component

### PR DESCRIPTION
**Pull request by erwinmombay**
still bikeshedding new property name

### CODE REVIEW

**Comment by erwinmombay** [comment]
PTAL @aghassemi

File: examples/live-list-update-reverse.amp.html [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      display: inline;
+      border: 1px solid black;
+      padding: 10px;
+      list-style-type: none;
+    }
+  </style>
+  <link href='https://fonts.googleapis.com/css?family=Georgia|Open+Sans|Roboto' rel='stylesheet' type='text/css'>
+  <!-- NOTE: This script tag makes this document not a valid AMP but is needed to turn on the experiment feature -->
+  <script>
+      (self.AMP=self.AMP||[]).push(function(AMP){
```
Comment (aghassemi) [inline examples/live-list-update-reverse.amp.html]: without refreshing the page this won't do much on the first try. I would say just make a "Turn sort experiment on" button.

File: extensions/amp-live-list/0.1/amp-live-list.js [hunk 2]
```diff
@@ -352,6 +359,10 @@ export class AmpLiveList extends AMP.BaseElement {
 
     if (hasInsertItems) {
       promise = promise.then(() => {
+        if (this.isReverseOrder_ && this.itemsSlot_.lastElementChild) {
```
Comment (aghassemi) [inline extensions/amp-live-list/0.1/amp-live-list.js]: nit: let's do one call to `this.viewport_.animateScrollIntoView` and move the logic to assignment of a `elementToScrollTo` variable

File: extensions/amp-live-list/0.1/amp-live-list.js [hunk 3]
```diff
@@ -387,17 +398,48 @@ export class AmpLiveList extends AMP.BaseElement {
...
-          // can be the last item.
-          } else if (!child.nextElementSibling) {
-            this.itemsSlot_.appendChild(orphan);
+
+        if (this.isReverseOrder_) {
+          for (let child = this.itemsSlot_.lastElementChild; child;
+            child = child.previousElementSibling) {
+            const childSortTime = this.getSortTime_(child);
+            if (orphanSortTime >= childSortTime) {
+              if (child.nextElementSibling) {
```
Comment (aghassemi) [inline extensions/amp-live-list/0.1/amp-live-list.js]: `insertAfterOrAtStart` from `dom.js` would do this.

File: extensions/amp-live-list/0.1/amp-live-list.js [hunk 4]
```diff
@@ -861,6 +921,11 @@ export class AmpLiveList extends AMP.BaseElement {
         this.viewport_.getScrollTop() + this.viewport_.getSize().height;
   }
 
+  isElementAboveViewport_(element) {
+    return this.viewport_.getLayoutRect(element).bottom <
+        this.viewport_.getScrollTop();
```
Comment (aghassemi) [inline extensions/amp-live-list/0.1/amp-live-list.js]: Not sure if `getScrollTop()` would be accurate here ( given the whole padding-top we add for viewer header). I suggest either:
1- do `viewport.getRect().bottom < element.getLayoutBox().top` or
2- check if `element.getIntersectionChangeEntry()` has `intersectionRatio` of 1.
(I assume element is an amp element?)

Comment (erwinmombay) [reply extensions/amp-live-list/0.1/amp-live-list.js]: @aghassemi element here can be a normal html tag (doesn't have to be a component) so i would probably need to do option 1.

Comment (aghassemi) [reply extensions/amp-live-list/0.1/amp-live-list.js]: sgtm

File: extensions/amp-live-list/0.1/amp-live-list.js (hunk 3 above)
Comment (erwinmombay) [reply extensions/amp-live-list/0.1/amp-live-list.js]: its 90% of what I need but I need the logic of "keep retrying until there is no more previousElementSibling" and breaking early

File: extensions/amp-live-list/0.1/amp-live-list.js (hunk 2 above)
Comment (erwinmombay) [reply extensions/amp-live-list/0.1/amp-live-list.js]: done.

File: extensions/amp-live-list/0.1/amp-live-list.js (hunk 4 above)
Comment (erwinmombay) [reply extensions/amp-live-list/0.1/amp-live-list.js]: @aghassemi `element.getLayoutBox()` doesn't exist correct?

File: examples/live-list-update-reverse.amp.html (hunk 1 above)
Comment (erwinmombay) [reply examples/live-list-update-reverse.amp.html]: @aghassemi can you explain why this wouldn't work on first load? I double checked and it seems to work. This script tag is non async and will be the first executed callback in the AMP pipeline so before amp-live-list even get initialized this experiment should have been set on the cookie already. LMK if i'm misunderstanding something or checked it wrong.

**Comment by cramforce** [comment]
This pull request **introduces 1 alert** when merging 4e499a45497e3251b6ceb59f246b72b916bb3fe2 into 7d9485dc0e80379f380f4f29b7094ebfb3f31c6d - [view on lgtm.com](https://lgtm.com/projects/g/ampproject/amphtml/rev/pr-e06b6aa5536b53eedc7b2dd2f5eeab1516033b95)

**new alerts:**

* 1 for Syntax error

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Comment by erwinmombay** [comment]
@aghassemi PTAL. would like to get this into tomorrows canary =X

File: examples/live-list-update-reverse.amp.html (hunk 1 above)
Comment (aghassemi) [reply examples/live-list-update-reverse.amp.html]: nvm me , didn't realize it is `sync`

**Pull request merged**